File: content/polympiads.2/3.prochain-k-arret/.codes/empty.cpp
#include <iostream>
#include <vector>

using namespace std;

int main () {
    int nbArrets;
    cin >> nbArrets;

    vector<int> parents(nbArrets);
    for (int i = 0; i < nbArrets; i ++) {
        cin >> parents[i];
    }

    int nbRestaurants;
    cin >> nbRestaurants;

    vector<int> distancesALArret (nbRestaurants);
    vector<int> arretDepart (nbRestaurants);
    for (int i = 0; i < nbRestaurants; i ++) {
        cin >> distancesALArret[i] >> arretDepart[i];
    }
    
    // Affichez pour chaque requête l'arrêt à une distance distancesALArret[i] de l'arrêt arretDepart[i]
}

File: content/polympiads.2/3.prochain-k-arret/.codes/empty.py
nb_arrets = int(input())
parents = list(map(int, input().split()))
nb_restaurants = int(input())
distances_a_l_arret = []
arret_depart = []
for i in range(nb_restaurants):
    k, d = list(map(int, input().split()))
    distances_a_l_arret.append(k)
    arret_depart.append(d)

# Affichez pour chaque requête l'arrêt à une distance distances_a_l_arret[i] de l'arrêt arret_depart[i]